### ISSUE TITLE
Todo app step 29 refers to 'modal' in reference to a regular form being shown/hidden

### ISSUE THREAD

**Issue by hbar1st** [freeCodeCamp/freeCodeCamp#55456]
### Describe the Issue

Step 29 of the Todo app (js beta) project says: "Step 29
After adding the task to the page, you should close the form modal to view the task. To do this, utilize classList to toggle the hidden class on the taskForm element." But is the form here really a 'modal'? I found that word confusing in this sentence considering that  the same project talks about modals in a different context.
Ref to the first time modal is discussed: https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-localstorage-by-building-a-todo-app/step-7
I think a better sentence should have said "hide the form" instead of "close the form modal".

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-localstorage-by-building-a-todo-app/step-29

### Your code

N/A

### Expected behavior

N/A

### System

N/A

**Comment by lasjorg** [comment]
Not sure if it qualifies as a modal.

Here is MDNs definition of modal and non-modal from the [dialog](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog) element docs.

> The HTML `<dialog>` element is used to create both modal and non-modal dialog boxes. Modal dialog boxes interrupt interaction with the rest of the page being inert, while non-modal dialog boxes allow interaction with the rest of the page.

I guess technically, the form does block the interaction with the part of the page that is used for adding a new task. But visually, it doesn't really look much like a modal.

**Comment by hbar1st** [comment]
agreed but also as I noted earlier, the same project doesn't define modal in a way that would apply to the form.

**Comment by lasjorg** [comment]
A modal isn't a specific element. I don't think calling it a "form modal" is wrong. It's a form, and it blocks the interaction with the other page elements.

As for step 7 in the todo app. It is a modal dialog box, it is a dialog box and it blocks the interaction.

---

I think both steps are correct, I'm just not sure if we ever give a definition of what a modal is.

**Comment by hbar1st** [comment]
yes there is no specific definition but as the only reference to 'modal' is given in step 7, it is normal to associate further uses of the word to that concept. (even if it is doesn't apply or isn't intended).
Maybe the word modal should be defined back in step 7 then?

**Comment by naomi-lgbt** [comment]
A PR which resolves this issue will update the beginning of the instructions in step 7 to be something like:

> A <dfn>modal</dfn> is an element that prevents all interaction with elements outside it until the modal has been dismissed.
>
> The HTML dialog element has a showModal() method that can be used to display a modal dialog box on a web page.

Happy to iterate over the wording further in a PR review, but let's get this unblocked and open for contribution. 🙂
